### PR TITLE
fix(auth-admin): Move sleep logic from messgeHandler to worker service

### PR DESCRIPTION
**Pull request by magnearun**
## What

Handle sleep logic in worker instead of messageHandler

## Why
Better separation of concerns and 
when the worker manages sleep, it doesn’t pull new messages during non-working hours, which optimizes resource usage and reduces unnecessary processing.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Release Notes

- **New Features**
  - Introduced a configuration option for notifications to operate continuously without pauses for non-working hours.
  - Added new utility functions for time management, including checks for working hours and sleep duration calculations.

- **Improvements**
  - Enhanced the `WorkerService` to manage operations based on working hours, allowing for better control and efficiency.
  - Updated the `NotificationsWorkerService` to simplify notification processing by removing time-based checks.

- **Tests**
  - Added new test cases to validate the functionality of time-related utilities and ensure robust notification handling.

These updates streamline notification processing and improve overall system responsiveness.

### CODE REVIEW

File: libs/message-queue/src/lib/nest/worker.service.ts [hunk 1]
```diff
@@ -46,7 +47,19 @@ export class WorkerService implements OnModuleDestroy {
     )
 
     this.status = Status.Running
+
     while (this.status === Status.Running) {
+      const now = new Date()
```
Comment (eirikurn) [inline libs/message-queue/src/lib/nest/worker.service.ts:52]: ~Shouldn't need to pass now date to utils. Tests can mock the time with useFakeTimers.

File: libs/message-queue/src/lib/nest/worker.service.ts [hunk 2]
```diff
@@ -46,7 +47,19 @@ export class WorkerService implements OnModuleDestroy {
...
+      if (
+        this.config.shouldSleepOutsideWorkingHours &&
+        isOutsideWorkingHours(now)
+      ) {
+        this.logger.info('Outside of working hours - worker sleeping')
```
Comment (eirikurn) [inline libs/message-queue/src/lib/nest/worker.service.ts:57]: ~Should add another log when we wake up? 🤷

File: libs/message-queue/src/lib/nest/worker.service.ts [hunk 3]
```diff
@@ -46,7 +47,19 @@ export class WorkerService implements OnModuleDestroy {
     )
 
     this.status = Status.Running
+
     while (this.status === Status.Running) {
+      const now = new Date()
+      if (
```
Comment (eirikurn) [inline libs/message-queue/src/lib/nest/worker.service.ts:53]: ~Can move this whole condition inside something like?

```
await this.maybeSleepOutsideWorkingHours()
```

**Review by eirikurn — APPROVED** [review]
Good stuff. Just nitpicks. 👍

**Review by rafnarnason — APPROVED** [review]
lgtm

**Pull request merged**